modules/personalize_taxonomy_context/plugins/visitor_context/TaxonomyContext.inc

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @file
+ * Provides a visitor context plugin for taxonomy.
+
+ */
+
+class TaxonomyContext extends PersonalizeContextBase {
+  /**
+   * Implements PersonalizeContextInterface::create().
+   */
+  public static function create(PersonalizeAgentInterface $agent = NULL, $selected_context = array()) {
+    return new self($agent, $selected_context);
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getOptions().
+   */
+  public static function getOptions() {
+    $vocabularies = taxonomy_get_vocabularies();
+    $options = array();
+    foreach ($vocabularies as $vocabulary) {
+      $machine_name = $vocabulary->machine_name;
+      $options[$machine_name] = array(
+        'name' => $vocabulary->name,
+        'group' => 'Taxonomy',
+        'id' => $vocabulary->vid,
+      );
+    }
+    return $options;
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getPossibleValues().
+   */
+  public function getPossibleValues($limit = FALSE) {
+    $possible_values = array();
+    $options = $this->getOptions();
+    $free_tagging_vocabulary_machine_names = SELF::getFreeTaggingVocabulariesMachineNames();
```

Comment:
As discussed on Slack, we will allow free-tagging vocabulary, and limit the number of available terms to 100. [Updated](https://github.com/acquia/personalize_module/commit/9d0071f007e4ebfc64420d818053e5e0ea9ba3fe).